development/tools/pkg/rendertemplates/genericcomponent.go

```diff
@@ -0,0 +1,95 @@
+package rendertemplates
+
+import (
+	"strings"
+)
+
+// changeExtraRefsBase changes base_ref to base string for each extra_ref
+func (j *Job) changeExtraRefsBase(base string) {
+	if j.JobConfig["extra_refs"] != nil {
+		for extraRefIndex := range j.JobConfig["extra_refs"].(map[interface{}]interface{}) {
+			j.JobConfig["extra_refs"].(map[interface{}]interface{})[extraRefIndex].([]interface{})[0].(map[interface{}]interface{})["base_ref"] = base
+		}
+	}
+}
+
+// GenerateComponentJobs generates jobs for components
+func (r *RenderConfig) GenerateComponentJobs(global map[string]interface{}) {
+	if present := len(r.JobConfigs); present > 0 {
+		for repoIndex, repo := range r.JobConfigs {
+			var jobs []Job
+			hasComponentJobs := false
+
+			for _, job := range repo.Jobs {
+				// check if the jobConfig is in expected for component jobs format
+				if job.JobConfig["name"] == nil && job.JobConfig["path"] != nil {
+					hasComponentJobs = true
+					// generate component jobs
+
+					repository := strings.Split(repo.RepoName, "/")[2]
+					nameSuffix := repository + "-" + strings.Replace(job.JobConfig["path"].(string), "/", "-", -1)
+
+					// generate pre- and post-submit jobs for the next release
+					if ReleaseMatches(global["nextRelease"], job.jobConfigPre["release_since"], job.jobConfigPre["release_until"]) {
```

Comment:
Can we search for release_since and release_until in whole JobConfig not only in jobConfigPre and jobConfigPost? If you want pre and post generate for same set of release which I think is our case, you can set the boundaries in jobConfig to apply for both instead setting them twice for pre and post.